src/fstab-generator/fstab-generator.c

```diff
@@ -119,6 +123,11 @@ static int add_swap(
                 return 0;
         }
 
+        if (arg_sysroot_check) {
+                log_info("%s should be enabled in the initrd, will request daemon-reload.", what);
+                return true;
+        }
```

Comment:
Hmm, I think it would be a tiny bit sexier if we'd return true in this whole function even if arg_sysroot_check is not set, and we just wrote out a unit. That way the return code semantics are always the same, regardless in which mode we are invoked, i.e. regardless if we are called in the test or the generator mode.

Reply:
i.e. just change the `return 0` at the very end of the call to `return true;`